### PR TITLE
Fix warnings found with new analyzer CA1860: Prefer Length/Count/IsEmpty property check over Any()

### PR DESCRIPTION
**Pull request by buyaa-n**
- Fix warnings found with the [new analyzer CA1860](https://github.com/dotnet/roslyn-analyzers/pull/6236): `Prefer Length/Count/IsEmpty property check over Any()`, and updated its severity.

### CODE REVIEW

File: eng/CodeAnalysis.src.globalconfig [hunk 1]
```diff
@@ -468,6 +468,9 @@ dotnet_diagnostic.CA1858.severity = warning
 # CA1859: Use concrete types when possible for improved performance
 dotnet_diagnostic.CA1859.severity = warning
```
Comment (geeknoid) [inline eng/CodeAnalysis.src.globalconfig:469]: Do things get out of hand if you turn on analysis of internal stuff too?

File: src/libraries/System.Net.Http/src/System/Net/Http/HttpClientHandler.cs [hunk 2]
```diff
@@ -22,7 +22,7 @@ public partial class HttpClientHandler : HttpMessageHandler
     {
         private readonly HttpHandlerType _underlyingHandler;
 
-        private HttpMessageHandler Handler
+        private HttpHandlerType Handler
 #if TARGET_BROWSER
```
Comment (stephentoub) [inline src/libraries/System.Net.Http/src/System/Net/Http/HttpClientHandler.cs:26]: To fix the error, you might change this to be more like:
```C#
#if TARGET_BROWSER
    private HttpMessageHandler Handler { get; }
#else
    private HttpHandlerType Handler => _underlyingHandler;
#endif
```
(although this approach with TARGET_BROWSER in general should be revisited, but that pre-exists your change)

**Review by stephentoub — COMMENTED** [review]
Changes generally look good, but there are more places that need to be fixed as CI highlights.

File: eng/CodeAnalysis.src.globalconfig (hunk 1 above)
Comment (geeknoid) [reply eng/CodeAnalysis.src.globalconfig:469]: Yes, the default is private and you can opt-in to internal and public too.

Support for internal and public is in a PR that is about to be merged: https://github.com/dotnet/roslyn-analyzers/pull/6421

Comment (buyaa-n) [reply eng/CodeAnalysis.src.globalconfig:469]: Yes, I tried with internal a lot flagged that looks valid, now https://github.com/dotnet/roslyn-analyzers/pull/6421 is merged, probably better wait until it is ready for consumption for the repo before enable the internal API surface

**Review by marek-safar — APPROVED** [review]

**Pull request merged**